lib/vanity/adapters/active_record_adapter.rb

```diff
@@ -19,10 +19,20 @@ def self.needs_attr_accessible?
         end
 
         def self.rails_agnostic_find_or_create_by(method, value)
-          if respond_to? :find_or_create_by
-            find_or_create_by(method => value)
-          else
-            send :"find_or_create_by_#{method}", value
+          retried = false
+          begin
+            if respond_to? :find_or_create_by
+              find_or_create_by(method => value)
+            else
+              send :"find_or_create_by_#{method}", value
+            end
+          rescue ActiveRecord::RecordNotUnique
+            if retried
+              raise
+            else
+              retried = true
+              retry
```

Comment:
Why would we retry after the first not unique? Won't the retry also be non unique?

Reply:
Assume processes A, B both call this code for the same experiment. In both processes find_or_create does not find an existing row and tries to create one.

A successfully inserts row into db, b fails with RecordNotUnique. B retries, this time find_or_create returns the row created by A

Reply:
🤦‍♂️ derp. Right.